src/Core__Promise.resi

```diff
@@ -19,13 +28,27 @@ let p = Promise.resolve(5) // promise<int>
 @scope("Promise")
 external resolve: 'a => t<'a> = "resolve"
 
+/**
+`reject(exn)` reject a Promise.
+See [`Promise.reject`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/reject) on MDN.
+
+## Examples
+
+```rescript
+let p = Promise.make((_, reject) => {
+  reject(. #error)
```

Comment:
Maybe change this to use a custom defined exception, just to make it clear it's indeed an exception and that you can do that?